### PR TITLE
GSP-127: Add ServiceInternel and RequestThrottled Errors

### PR DESCRIPTION
**Pull request by xxchan**
Part of (abandoned) #60

### CODE REVIEW

File: rfcs/127-add-serverinternel-and-requestthrottled-errors.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# GSP-127: Add ServiceInternel and RequestThrottled Errors
+
+## Background
+
+These are common errors, especially in HTTP-based services, but also apply to RPC-based services. And they are retryable errors. If we provide them, users can retry only them instead of all errors.
+
+## Proposal
+
+I propose to add the following global error codes:
+- `ErrServerInternal`: e.g., HTTP 5xx
```
Comment (Xuanwo) [inline rfcs/127-add-serverinternel-and-requestthrottled-errors.md:15]: Maybe `ServiceInternel`?

File: rfcs/127-add-serverinternel-and-requestthrottled-errors.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- `ErrRequestThrottled`: e.g., HTTP 429 Too Many Requests/Limit Exceeded, 503 SlowDown, ...
+
+## Rationale
+
+Alternative names
+- `ErrServer`/`ErrInternal` & `ErrThrottling`/`ErrThrottled`: short, but not conform to `Err<Noun><Predicate>`
+- `ErrServerThrottling`: maybe not as common as `RequestThrottled`?
```
Comment (Xuanwo) [inline rfcs/127-add-serverinternel-and-requestthrottled-errors.md:22]: How about `ServiceRequestThrottled`? So that we can make the scope more clear?

Comment (xxchan) [reply rfcs/127-add-serverinternel-and-requestthrottled-errors.md:22]: I think "service request" seems strange

File: rfcs/127-add-serverinternel-and-requestthrottled-errors.md (hunk 1 above)
Comment (xxchan) [reply rfcs/127-add-serverinternel-and-requestthrottled-errors.md:15]: I can accept it

File: rfcs/127-add-serverinternel-and-requestthrottled-errors.md (hunk 2 above)
Comment (Xuanwo) [reply rfcs/127-add-serverinternel-and-requestthrottled-errors.md:22]: OK, `RequestThrottled` looks good to me.

**Comment by Xuanwo** [comment]
ping @JinnyYi for review.

File: rfcs/127-add-serverinternel-and-requestthrottled-errors.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+# GSP-127: Add ServiceInternel and RequestThrottled Errors
+
+## Background
+
+These are common errors, especially in HTTP-based services, but also apply to RPC-based services. And they are retryable errors. If we provide them, users can retry only them instead of all errors.
+
+## Proposal
+
+I propose to add the following global error codes:
```
Comment (JinnyYi) [inline rfcs/127-add-serverinternel-and-requestthrottled-errors.md:14]: Is it necessary to add error types to point out the error code or error message?

**Review by Xuanwo — APPROVED** [review]

File: rfcs/127-add-serverinternel-and-requestthrottled-errors.md (hunk 3 above)
Comment (xxchan) [reply rfcs/127-add-serverinternel-and-requestthrottled-errors.md]: I think we don't need to do so now. `fmt.Errorf("%w: %v", ErrServiceInternal, err) is enough.

**Review by JinnyYi — APPROVED** [review]

**Pull request merged**